tests/ignite/metrics/test_classification_report.py

```diff
@@ -0,0 +1,297 @@
+        y_true_unflat = _unflatten_binary(y_true)
+        classification_report = ClassificationReport(device=metric_device, output_dict="False", labels=labels)
+
+        def update(engine, i):
+            return (
+                y_true_unflat[i * s + rank * offset : (i + 1) * s + rank * offset],
+                y_preds[i * s + rank * offset : (i + 1) * s + rank * offset],
+            )
+
+        engine = Engine(update)
+
+        classification_report.attach(engine, "cr")
+
+        data = list(range(n_iters))
+        engine.run(data=data)
+
+        assert "cr" in engine.state.metrics
+        res = engine.state.metrics["cr"]
+        res2 = classification_report.compute()
+        assert res == res2
+
+        from sklearn.metrics import classification_report as sklearn_classification_report
+
+        sklearn_result = sklearn_classification_report(y_preds.cpu().numpy(), y_true.cpu().numpy(), output_dict=True)
+
+        assert pytest.approx(res[labels[0]] == sklearn_result["0"])
+        assert pytest.approx(res[labels[1]] == sklearn_result["1"])
+        assert pytest.approx(res["macro avg"]["precision"] == sklearn_result["macro avg"]["precision"])
+        assert pytest.approx(res["macro avg"]["recall"] == sklearn_result["macro avg"]["recall"])
+        assert pytest.approx(res["macro avg"]["f1-score"] == sklearn_result["macro avg"]["f1-score"])
+
+    for _ in range(5):
+        # check multiple random inputs as random exact occurencies are rare
+        metric_devices = ["cpu"]
+        if device.type != "xla":
+            metric_devices.append(idist.device())
+        for metric_device in metric_devices:
+            _test(metric_device)
+
+
+def _test_integration_multilabel(device, output_dict):
+
+    rank = idist.get_rank()
+    torch.manual_seed(12)
+
+    def _test(metric_device, n_classes):
+
+        n_iters = 80
+        s = 16
+        offset = n_iters * s
+        y_true = torch.randint(0, n_classes, size=(offset * idist.get_world_size(),)).to(device)
+        y_preds = torch.randint(0, n_classes, size=(offset * idist.get_world_size(),)).to(device)
+        y_true_unflat = _unflatten_multilabel(y_true, n_classes)
+        classification_report = ClassificationReport(device=metric_device, output_dict=output_dict)
+
+        def update(engine, i):
+            return (
+                y_true_unflat[i * s + rank * offset : (i + 1) * s + rank * offset],
+                y_preds[i * s + rank * offset : (i + 1) * s + rank * offset],
+            )
+
+        engine = Engine(update)
+
+        classification_report.attach(engine, "cr")
+
+        data = list(range(n_iters))
+        engine.run(data=data)
+
+        assert "cr" in engine.state.metrics
+        res = engine.state.metrics["cr"]
+        res2 = classification_report.compute()
+        assert res == res2
+
+        assert isinstance(res, dict if output_dict else str)
+        if not output_dict:
+            res = ast.literal_eval(res)
```

Comment:
Can you please explain why we need to use ast here ?

Reply:
I've replaced it with the more standard `json` module